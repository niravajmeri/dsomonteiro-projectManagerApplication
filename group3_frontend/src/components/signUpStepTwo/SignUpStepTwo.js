import React, { Component } from 'react';
import { Button, ButtonToolbar } from 'react-bootstrap';

const SMSVALIDATION = 'smsValidation';
const EMAILVALIDATION = 'emailValidation';

class SignUpStepTwo extends Component {
    render() {
        return this.stepTwoValidation();
    }

    stepTwoValidation = (title, i) => (
        <div className="stepTwoValidation">
            <div className="chooseValidationMethodText">
                To continue with sign up process, you will receive a validation
                code by SMS or email. Please, choose which one:
            </div>
            <ButtonToolbar className="buttonValidationToolbar">
                <Button
                    id="smsValidation"
                    onClick={this.handleSignUpValidation}
                >
                    SMS
                </Button>
                <Button
                    id="emailValidation"
                    onClick={this.handleSignUpValidation}
                >
                    Email
                </Button>
            </ButtonToolbar>
        </div>
    );

    handleSignUpValidation = event => {
        const validationType = event.target.id;

        if (
            validationType === SMSVALIDATION ||
            validationType === EMAILVALIDATION
        ) {
            return fetch(this.props.validationUrl[validationType])
                .then(response => {
                    return response.json();
                })
                .then(jsonResponse => {
                    this.props.setStepTwo({
                        signUpStep: 3,
                        verificationUrl: jsonResponse.href
                    });
                });
        }
        return;
    };
}

export default SignUpStepTwo;
