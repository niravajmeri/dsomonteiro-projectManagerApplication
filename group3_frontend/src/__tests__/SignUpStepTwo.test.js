import React from 'react';
import { shallow } from 'enzyme';
import SignUpStepTwo from '../components/signUpStepTwo/SignUpStepTwo';

describe('SignUpStepTwo', () => {
    const fetchResult = {
        json: jest.fn().mockReturnValue({ href: 'responseUrl' })
    };
    global.fetch = jest
        .fn()
        .mockImplementation(() => Promise.resolve(fetchResult));

    let mockedClass;

    afterEach(() => {
        global.fetch.mockClear();
    });

    beforeEach(() => {
        mockedClass = new SignUpStepTwo();
    });

    it('should fetch data with smsValidation validation type', () => {
        const eventSmsMock = {
            target: {
                id: 'smsValidation'
            }
        };

        mockedClass.props = {
            validationUrl: {
                smsValidation: 'smsValidationUrl'
            },
            setStepTwo: jest.fn()
        };
        mockedClass.handleSignUpValidation(eventSmsMock).then(() => {
            const expectedProps = {
                signUpStep: 3,
                verificationUrl: 'responseUrl'
            };
            expect(mockedClass.props.setStepTwo).toHaveBeenCalledWith(
                expectedProps
            );
        });
        expect(global.fetch).toHaveBeenCalledWith(
            mockedClass.props.validationUrl.smsValidation
        );
    });

    it('should fetch data with emailValidation validation type', () => {
        const eventSmsMock = {
            target: {
                id: 'emailValidation'
            }
        };

        mockedClass.props = {
            validationUrl: {
                emailValidation: 'emailValidationUrl'
            },
            setStepTwo: jest.fn()
        };
        mockedClass.handleSignUpValidation(eventSmsMock).then(() => {
            const expectedProps = {
                signUpStep: 3,
                verificationUrl: 'responseUrl'
            };
            expect(mockedClass.props.setStepTwo).toHaveBeenCalledWith(
                expectedProps
            );
        });
        expect(global.fetch).toHaveBeenCalledWith(
            mockedClass.props.validationUrl.emailValidation
        );
    });

    it('should render correctly', () => {
        const component = shallow(<SignUpStepTwo />);
        expect(component).toMatchSnapshot();
    });

    it('should not fetch data when receives a invalid validation type', () => {
        const eventSmsMock = {
            target: {
                id: 'invalid'
            }
        };

        mockedClass.handleSignUpValidation(eventSmsMock);

        expect(global.fetch).not.toHaveBeenCalled();
    });
});
