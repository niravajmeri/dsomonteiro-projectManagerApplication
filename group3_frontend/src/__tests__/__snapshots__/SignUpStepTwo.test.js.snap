// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignUpStepTwo should render correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SignUpStepTwo />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="chooseValidationMethodText"
>
          To continue with sign up process, you will receive a validation code by SMS or email. Choose the method:
</div>,
        <ButtonToolbar
          bsClass="btn-toolbar"
          className="buttonValidationToolbar"
>
          <Button
                    active={false}
                    block={false}
                    bsClass="btn"
                    bsStyle="default"
                    disabled={false}
                    id="smsValidation"
                    onClick={[Function]}
          >
                    SMS
          </Button>
          <Button
                    active={false}
                    block={false}
                    bsClass="btn"
                    bsStyle="default"
                    disabled={false}
                    id="emailValidation"
                    onClick={[Function]}
          >
                    Email
          </Button>
</ButtonToolbar>,
      ],
      "className": "stepTwoValidation",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "To continue with sign up process, you will receive a validation code by SMS or email. Choose the method:",
          "className": "chooseValidationMethodText",
        },
        "ref": null,
        "rendered": "To continue with sign up process, you will receive a validation code by SMS or email. Choose the method:",
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "bsClass": "btn-toolbar",
          "children": Array [
            <Button
              active={false}
              block={false}
              bsClass="btn"
              bsStyle="default"
              disabled={false}
              id="smsValidation"
              onClick={[Function]}
>
              SMS
</Button>,
            <Button
              active={false}
              block={false}
              bsClass="btn"
              bsStyle="default"
              disabled={false}
              id="emailValidation"
              onClick={[Function]}
>
              Email
</Button>,
          ],
          "className": "buttonValidationToolbar",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "active": false,
              "block": false,
              "bsClass": "btn",
              "bsStyle": "default",
              "children": "SMS",
              "disabled": false,
              "id": "smsValidation",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "SMS",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "active": false,
              "block": false,
              "bsClass": "btn",
              "bsStyle": "default",
              "children": "Email",
              "disabled": false,
              "id": "emailValidation",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "Email",
            "type": [Function],
          },
        ],
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="chooseValidationMethodText"
>
            To continue with sign up process, you will receive a validation code by SMS or email. Choose the method:
</div>,
          <ButtonToolbar
            bsClass="btn-toolbar"
            className="buttonValidationToolbar"
>
            <Button
                        active={false}
                        block={false}
                        bsClass="btn"
                        bsStyle="default"
                        disabled={false}
                        id="smsValidation"
                        onClick={[Function]}
            >
                        SMS
            </Button>
            <Button
                        active={false}
                        block={false}
                        bsClass="btn"
                        bsStyle="default"
                        disabled={false}
                        id="emailValidation"
                        onClick={[Function]}
            >
                        Email
            </Button>
</ButtonToolbar>,
        ],
        "className": "stepTwoValidation",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "To continue with sign up process, you will receive a validation code by SMS or email. Choose the method:",
            "className": "chooseValidationMethodText",
          },
          "ref": null,
          "rendered": "To continue with sign up process, you will receive a validation code by SMS or email. Choose the method:",
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "bsClass": "btn-toolbar",
            "children": Array [
              <Button
                active={false}
                block={false}
                bsClass="btn"
                bsStyle="default"
                disabled={false}
                id="smsValidation"
                onClick={[Function]}
>
                SMS
</Button>,
              <Button
                active={false}
                block={false}
                bsClass="btn"
                bsStyle="default"
                disabled={false}
                id="emailValidation"
                onClick={[Function]}
>
                Email
</Button>,
            ],
            "className": "buttonValidationToolbar",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "active": false,
                "block": false,
                "bsClass": "btn",
                "bsStyle": "default",
                "children": "SMS",
                "disabled": false,
                "id": "smsValidation",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "SMS",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "active": false,
                "block": false,
                "bsClass": "btn",
                "bsStyle": "default",
                "children": "Email",
                "disabled": false,
                "id": "emailValidation",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "Email",
              "type": [Function],
            },
          ],
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
